### PR TITLE
Gpl 744 as a developer i would like to generate multiple ht barcodes for beckman destination plates

### LINKED ISSUE #60: GPL-744 - As a developer I would like to generate multiple HT barcodes for beckman destination plates
**User story**
As a developer I would like to generate HT barcodes for beckman destination plates.

**Who are the primary contacts for this story**
Steve / Harry

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] I should be able to send a request for multiple HT barcodes and receive an array of barcodes
- [ ] ...

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by stevieing**
Closes #60 

Changes proposed in this pull request:

*
*
* ...

### CODE REVIEW

**Comment by stevieing** [comment]
I have done enough to get this working. There could be more refactoring but I needed to stop somewhere to get printing again.

Be gentle! A lot of work has gone into this the past couple of days.

TODO:
- Add some comments
- Add better type checking
- Add better object passing e.g. creating heron formatter in operations
- Add more tests for existing code

File: Pipfile [hunk 1]
```diff
@@ -17,6 +17,7 @@ sqlalchemy-stubs = "*"
 flask = "~=1.1"
```
Comment (pjvv) [inline Pipfile:17]: Can you run `pipenv update` too since we haven't touched this project in a while.

File: baracoda/helpers.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+logger = logging.getLogger(__name__)
+
+from flask import current_app
+
+def get_prefix_item(prefix: str):
+  """
+    Method which returns prefix object.
+    This will contain the prefix, the sequence and whether any barcode for that prefix needs
+    toi be formatted.
```
Comment (pjvv) [inline baracoda/helpers.py:11]:
```suggestion
    to be formatted.
```

File: baracoda/helpers.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    Arguments:
+        value {str} -- the actual prefix
+
+    Returns:
+        str -- prefix object e.g. { "prefix": "abc", "sequence_name": "seq", "convert": False}
```
Comment (pjvv) [inline baracoda/helpers.py:17]: This seems to be returning a Dict or None?

File: baracoda/helpers.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+from flask import current_app
+
+def get_prefix_item(prefix: str):
```
Comment (pjvv) [inline baracoda/helpers.py:7]: Can add the type here which would be helpful.

File: baracoda/operations.py [hunk 5]
```diff
@@ -8,19 +8,30 @@
...
         self.__check_prefix()
 
-        self.formatter = HeronFormatter(prefix=self.prefix)
+        self.__set_prefix_item()
+
+        # if the prefix item does not exist the prefix is not valid
+        if not self.prefix_item:
+            raise InvalidPrefixError()
+        
+        # saves pulling it out of object every time
```
Comment (pjvv) [inline baracoda/operations.py:31]: Maybe adding a check that the keys are available in the dict would be safe? Someone could forget to add them in the config.

Comment (stevieing) [reply baracoda/operations.py:31]: Added check to config.py as per discussion.

**Review by pjvv — APPROVED** [review]

**Pull request merged**